infra/feast-operator/internal/controller/auth/auth.go

```diff
@@ -0,0 +1,220 @@
+	role := auth.initFeastRole()
+	if op, err := controllerutil.CreateOrUpdate(auth.Handler.Context, auth.Handler.Client, role, controllerutil.MutateFn(func() error {
+		return auth.setFeastRole(role)
+	})); err != nil {
+		return err
+	} else if op == controllerutil.OperationResultCreated || op == controllerutil.OperationResultUpdated {
+		logger.Info("Successfully reconciled", "Role", role.Name, "operation", op)
+	}
+
+	return nil
+}
+
+func (auth *FeastAuth) initFeastRole() *rbacv1.Role {
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{Name: auth.getFeastRoleName(), Namespace: auth.Handler.FeatureStore.Namespace},
+	}
+	role.SetGroupVersionKind(rbacv1.SchemeGroupVersion.WithKind("Role"))
+	return role
+}
+
+func (auth *FeastAuth) setFeastRole(role *rbacv1.Role) error {
+	role.Labels = auth.getLabels()
+	role.Rules = []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{rbacv1.GroupName},
+			Resources: []string{"roles", "rolebindings"},
+			Verbs:     []string{"get", "list", "watch"},
+		},
+	}
+
+	return controllerutil.SetControllerReference(auth.Handler.FeatureStore, role, auth.Handler.Scheme)
+}
+
+func (auth *FeastAuth) createFeastRoleBinding() error {
+	logger := log.FromContext(auth.Handler.Context)
+	roleBinding := auth.initFeastRoleBinding()
+	if op, err := controllerutil.CreateOrUpdate(auth.Handler.Context, auth.Handler.Client, roleBinding, controllerutil.MutateFn(func() error {
+		return auth.setFeastRoleBinding(roleBinding)
+	})); err != nil {
+		return err
+	} else if op == controllerutil.OperationResultCreated || op == controllerutil.OperationResultUpdated {
+		logger.Info("Successfully reconciled", "RoleBinding", roleBinding.Name, "operation", op)
+	}
+
+	return nil
+}
+
+func (auth *FeastAuth) initFeastRoleBinding() *rbacv1.RoleBinding {
+	roleBinding := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{Name: auth.getFeastRoleName(), Namespace: auth.Handler.FeatureStore.Namespace},
+	}
+	roleBinding.SetGroupVersionKind(rbacv1.SchemeGroupVersion.WithKind("RoleBinding"))
+	return roleBinding
+}
+
+func (auth *FeastAuth) setFeastRoleBinding(roleBinding *rbacv1.RoleBinding) error {
+	roleBinding.Labels = auth.getLabels()
+	roleBinding.Subjects = []rbacv1.Subject{}
+	if auth.Handler.FeatureStore.Status.Applied.Services.OfflineStore != nil {
+		roleBinding.Subjects = append(roleBinding.Subjects, rbacv1.Subject{
+			Kind:      rbacv1.ServiceAccountKind,
+			Name:      services.GetFeastServiceName(auth.Handler.FeatureStore, services.OfflineFeastType),
+			Namespace: auth.Handler.FeatureStore.Namespace,
+		})
+	}
+	if auth.Handler.FeatureStore.Status.Applied.Services.OnlineStore != nil {
+		roleBinding.Subjects = append(roleBinding.Subjects, rbacv1.Subject{
+			Kind:      rbacv1.ServiceAccountKind,
+			Name:      services.GetFeastServiceName(auth.Handler.FeatureStore, services.OnlineFeastType),
+			Namespace: auth.Handler.FeatureStore.Namespace,
+		})
+	}
+	if auth.Handler.FeatureStore.Status.Applied.Services.Registry != nil {
```

Comment:
should this only be configured if a local registry is being deployed? if the user is leveraging a remote registry, it would be on them to handle inter-ns/cluster rbac configs/perms for registry access.
```suggestion
	if servces.isLocalRegistry(auth.Handler.FeatureStore) {
```

Reply:
u r right, thanks!